File: python/swap_node_in_pairs.py
# no.24
from utils import ListNode

class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head or not head.next:
            return head
        sublist = head.next.next
        new_head = head.next
        head.next = self.swapPairs(sublist)
        new_head.next = head
        return new_head

File: C++/swap_node_in_pairs.cpp
//no.24

#include "linked_list.h"

class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        if((!head) || (!head->next)){
            return head;
        }
        ListNode* sublist = head->next->next;
        ListNode* newHead = head->next;
        head->next = swapPairs(sublist);
        newHead->next = head;
        return newHead;
    }
};
